rosbag2_compression/include/rosbag2_compression/compression_options.hpp

```diff
@@ -58,11 +58,11 @@ ROSBAG2_COMPRESSION_PUBLIC std::string compression_mode_to_string(CompressionMod
  */
 struct CompressionOptions
 {
-  std::string compression_format;
-  CompressionMode compression_mode;
-  uint64_t compression_queue_size;
+  std::string compression_format{};
+  CompressionMode compression_mode{CompressionMode::NONE};
+  uint64_t compression_queue_size{0};
   /// \brief // The number of compression threads
-  uint64_t compression_threads;
+  uint64_t compression_threads{5};
```

Comment:
how about putting `0` instead? that will set up with `std::thread::hardware_concurrency()`?

```suggestion
  uint64_t compression_threads{0};
```

Reply:
I agree with @fujitatomoya better way to put **0** and we will override with HW threads available here
https://github.com/ros2/rosbag2/blob/317286c98472205593d5b0a791d06fea42613ba3/rosbag2_compression/src/rosbag2_compression/sequential_compression_writer.cpp#L176-L183